pkgs/development/python-modules/symengine/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cython
+, cmake
+, symengine
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "symengine";
+  version = "0.3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "e86d13aadc9f765f2c5462da32950edd36d1a0a52dbfc96e766be3689957c04d";
+  };
+
+  postConfigure = ''
+    substituteInPlace setup.py \
+      --replace "\"cmake\"" "\"${cmake}/bin/cmake\""
+
+    substituteInPlace cmake/FindCython.cmake \
+      --replace "SET(CYTHON_BIN cython" "SET(CYTHON_BIN ${cython}/bin/cython"
```

Comment:
You can pass, `--define="CYTHON_BIN=${cython}/bin/cython"` to setup.py instead of patching.

Reply:
Great didn't know that you could do this!